test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -222,26 +226,39 @@ def test_step(self):
             m = torch.nn.Linear(1, 1)
             m.weight.data = torch.tensor([[1.0]])
             m.bias.data = torch.tensor([2.0])
+            m_zero = copy.deepcopy(m)
             m.to(self.device)
+            m_zero.to(self.device)
+
+            lr = 0.1
+            o = SGD(m.parameters(), lr=lr)
+            o_zero = ZeroRedundancyOptimizer(m_zero.parameters(), optimizer_class=SGD, lr=lr)
 
-            o = ZeroRedundancyOptimizer(m.parameters(), optimizer_class=SGD, lr=0.1)
             y = m(x)
             y.backward(x)
+            y_zero = m_zero(x)
+            y_zero.backward(x)
+
```

Comment:
I opted to not use `DDP` directly since I think this implementation better shows the substitutability of `torch.distributed.optim.ZeroRedundancyOptimizer` with `torch.optim.SGD`. Let me know if there is any issue with this.